src/effects/LensFlare.tsx

```diff
@@ -0,0 +1,187 @@
+        ['lensDirtTexture', new THREE.Uniform(lensDirtTexture)],
+        ['haloScale', new THREE.Uniform(haloScale)],
+        ['secondaryGhosts', new THREE.Uniform(secondaryGhosts)],
+        ['aditionalStreaks', new THREE.Uniform(aditionalStreaks)],
+        ['ghostScale', new THREE.Uniform(ghostScale)],
+        ['starBurst', new THREE.Uniform(starBurst)],
+        ['opacity', new THREE.Uniform(opacity)],
+      ]),
+    })
+  }
+
+  update(_renderer: any, _inputBuffer: any, deltaTime: number) {
+    const iTime = this.uniforms.get('iTime')
+    if (iTime) {
+      iTime.value += deltaTime
+    }
+  }
+}
+
+type LensFlareProps = ConstructorParameters<typeof LensFlareEffect>[0] & {
+  position?: THREE.Vector3
+  followMouse?: boolean
+}
+
+export const LensFlare = forwardRef<LensFlareEffect, LensFlareProps>(
+  ({ position = new THREE.Vector3(-25, 6, -60), followMouse = false, ...props }, ref) => {
+    const viewport = useThree(({ viewport }) => viewport)
+    const raycaster = useThree(({ raycaster }) => raycaster)
+    const pointer = useThree(({ pointer }) => pointer)
+    const { scene, camera } = useContext(EffectComposerContext)
+
+    const [projectedPosition] = useState(() => new THREE.Vector3())
+    const [mouse2d] = useState(() => new THREE.Vector2())
+
+    const effect = useMemo(() => new LensFlareEffect(props), [props])
+
+    useFrame((_, delta) => {
+      const uLensPosition = effect.uniforms.get('lensPosition')
+      const uOpacity = effect.uniforms.get('opacity')
+      if (!uLensPosition || !uOpacity) return
+
+      let target = 1
+
+      if (followMouse) {
+        uLensPosition.value.x = pointer.x
+        uLensPosition.value.y = pointer.y
+        target = 0
+      } else {
+        projectedPosition.copy(position).project(camera)
+        if (projectedPosition.z > 1) return
+
+        uLensPosition.value.x = projectedPosition.x
+        uLensPosition.value.y = projectedPosition.y
+
+        mouse2d.set(projectedPosition.x, projectedPosition.y)
+        raycaster.setFromCamera(mouse2d, camera)
+        const intersects = raycaster.intersectObjects(scene.children, true)
+        const object = intersects[0].object
+        if (object) {
+          if (object.userData?.lensflare === 'no-occlusion') {
+            target = 0
+          } else if (object instanceof THREE.Mesh) {
+            if (object.material.uniforms?._transmission?.value > 0.2) {
+              //Check for MeshTransmissionMaterial
+              target = 0.2
+            } else if (object.material._transmission && object.material._transmission > 0.2) {
+              //Check for MeshPhysicalMaterial with transmission setting
+              target = 0.2
+            } else if (object.material.transparent) {
+              // Check for OtherMaterials with transparent parameter
+              target = object.material.opacity
+            }
+          }
+        }
+      }
+
+      easing.damp(uOpacity, 'value', target, 0.07, delta)
```

Comment:
@ektogamat maybe we can make the `0.07` value as a `smoothTime` prop?

Reply:
Yes, thank you. Lets do it.